raiden/lightclient/handlers/light_client_message_handler.py

```diff
@@ -43,10 +45,11 @@ class LightClientMessageHandler:
     def store_light_client_protocol_message(cls, identifier: int, message: Message, signed: bool, payment_id: int,
                                             order: int, message_type: LightClientProtocolMessageType,
                                             wal: WriteAheadLog):
+        payment = wal.storage.get_light_client_payment(payment_id)
         return wal.storage.write_light_client_protocol_message(
             message,
             build_light_client_protocol_message(identifier, message, signed,
-                                                payment_id, order, message_type)
+                                                payment_id, order, message_type, payment.light_client_address)
```

Comment:
it was my understanding that we could have LC messages without a payment involved.

so if we don't have a payment, how do we set the `light_client_address` in cases other than this one?

Reply:
I have the same question.

Also, wouldn't it fail in such case? (because payment is None)

Reply:
Yes but the query for payments is using a JOIN to get the information about the client, we don't want to remove that since we don't know all the places where it's being used so i keep the light_client_address inside the query for payements to keep it working and to keep the simplicity when you query a payment to get the client who did that payment, so basically with the payment id you can get the client.

Reply:
Long story short, no it will never be None since if you have a payment id you have a client who did that payment and the query for the payment will provide that information.

Reply:
I understand.

But what do we do when we need to store a message that is not directly associated with a payment (e.g. settlement, register_secret).

Reply:
You put None on that field since it's optional, the relationship is like this:

Payment -> Light Client Protocol Message -> Client

In this case i use the relation between Payment and Light Client Protocol Message to get the Client. If you have a Payment you always have messages but not always that you have a message it's for a payment.

Reply:
You were right about the error, a fix it's already implemented.